target/linux/ipq40xx/base-files/etc/board.d/02_network

```diff
@@ -130,6 +136,10 @@ ipq40xx_setup_macs()
 		lan_mac=$(cat /sys/firmware/mikrotik/hard_config/mac_base)
 		label_mac="$lan_mac"
 		;;
+	netgear,rbr50|\
+	netgear,rbs50|\
+	netgear,srr60|\
+	netgear,srs60|\
```

Comment:
Can this be replaced by using an alias for U-boot to populate it like it populates the DSA master one?
Or NVMEM?

Reply:
I tried enabling all the ports but they all get the same lan mac address, so I don't know where to get the wan mac address.
I don't know if NVMEM will ever work because of the mmc. I tried something like this but didn't work:
```
&sdhci {
	status = "okay";

	pinctrl-0 = <&sd_pins>;
	pinctrl-names = "default";
	cd-gpios = <&tlmm 22 GPIO_ACTIVE_LOW>;
	vqmmc-supply = <&vqmmc>;
	non-removable;
	#address-cells = <1>;
	#size-cells = <0>;

	emmc@0 {
		compatible = "mmc-card";
		reg = <0>;

		partitions {
			compatible = "fixed-partitions";
			#address-cells = <1>;
			#size-cells = <1>;

			partition@584400 {
				label = "ART";
				reg = <0x584400 0x40000>;
				read-only;

				macaddr_art_6: macaddr@c {
					reg = <0xc 0x6>;
				};
			};
		};
	};
};

&swport1 {
	status = "okay";

	label = "wan";

	nvmem-cells = <&macaddr_art_6>;
	nvmem-cell-names = "mac-address";
};
```
Do you have any other suggestions?

Reply:
```
		partitions {
			#address-cells = <1>;
			#size-cells = <1>;

			partition-art {
				compatible = ""nvmem-cells";
				label = "ART";
				reg = <0x584400 0x40000>;
				read-only;

				macaddr_art_6: macaddr@c {
					reg = <0xc 0x6>;
				};
			};
		};
```

Can you test ?

Reply:
Unfortunately it doesn't work :disappointed:. The path `/sys/bus/nvmem/devices/` is always empty. Maybe the "nvmem-cells" is not available for emmc? Or I am just miss something in the `sdhci` node?
Unfortunately I found only one device with emmc defined in the dts but it doesn't use any partition subnode: https://github.com/openwrt/openwrt/blob/master/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-ncp-hg100-cellular.dts#L566

Reply:
NVMEM does not work on eMMC as far as I know as it uses partition tables which are stored on the device.